rwengine/src/render/TextRenderer.cpp

```diff
@@ -12,18 +12,22 @@
 #include "engine/GameData.hpp"
 #include "render/GameRenderer.hpp"
 
-int charToIndex(uint16_t g) {
+unsigned charToIndex(std::uint16_t g) {
     // Correct for the default font maps
     /// @todo confirm for JA / RU font maps
     return g - 32;
 }
 
-glm::vec4 indexToCoord(font_t font, int index) {
-    float x = int(index % 16);
-    float y = int(index / 16) + 0.01f;
-    float fontHeight = ((font == FONT_PAGER) ? 16.f : 13.f);
-    glm::vec2 gsize(1.f / 16.f, 1.f / fontHeight);
-    return glm::vec4(x, y, x + 1, y + 0.98f) * glm::vec4(gsize, gsize);
+static glm::vec4 indexToTexCoord(int index, const glm::u32vec2 &textureSize, const glm::u8vec2 &glyphOffset) {
+    static const unsigned TEXTURE_COLUMNS = 16;
+    const float x = index % TEXTURE_COLUMNS;
+    const float y = index / TEXTURE_COLUMNS;
+    // Add offset to avoid 'leakage' between adjacent glyphs
+    float s = (x * glyphOffset.x + 0.5f) / textureSize.x;
+    float t = (y * glyphOffset.y + 0.5f) / textureSize.y;
+    float p = ((x + 1) * glyphOffset.x - 1.5f) / textureSize.x;
+    float q = ((y + 1) * glyphOffset.y - 1.5f) / textureSize.y;
```

Comment:
Without offsets, glyphs in adjacent places leak in the current glyph.
This is visible with some glyphs of arial.
Has somebody another suggestion?